### ISSUE TITLE
Whitespace breaks use of conditional within event handler assignment in template

### ISSUE THREAD

**Issue by frenchroasted** [ractivejs/ractive#2894]
## Description:
This event handler works: on-click="{{#if true}}handler{{/if}}"
This event handler never gets assigned: on-click=" {{#if true}}handler{{/if}}"

When whitespace is present before the {{#if ...}} or after the {{/if}}, the even handler is not bound. Whitespace within the {{#if}}...{{/if}} is ok, and whitespace ok if there is no conditional at all, but using whitespace in the crack between the quote and the {{#if}} somehow confuses the parser.

Edit to mention: It doesn't bother me if whitespace is not allowed in this situation; the problem is that there is no exception or error of any kind from the parser or the runtime, even in debug mode. The event handler silently fails, making debugging this kind of problem very difficult in a complex template.

## Versions affected:
Known: 0.8.9, 0.8.11  Probably: 0.8.0 through 0.8.11

## Platforms affected:
All

## Reproduction:
JSFiddle: https://jsfiddle.net/frenchroasted/r07fphky/

## Code:
```
<h1>Event Handlers and Spaces (Ractive 0.8.11)</h1>
<p>
  If an event handler is declared with an internal conditional, any whitespace in the wrong place will break the event binding.
</p>
<main id="main"></main>

<script id="myTemplate" type="x-template">
	<div>
  	<input
    	type="button"
      value="This button works"
      on-click="{{#if true}} alertMe {{/if}}"/>
    It is safe to include whitespace inside a #if.<br/>
    <br/>
    <input
    	type="button"
      value="This button is broken"
      on-click=" {{#if true}}alertMe{{/if}}"
    />
    <input
    	type="button"
      value="This button is also broken"
      on-click="{{#if true}}alertMe{{/if}} "
    />
    It is NOT safe to use whitespace outside the #if. Both leading and trailing whitespace will break the event binding.<br/>
  </div>
</script>
```

```
var r = new Ractive({
		el: 'main',
    template: '#myTemplate',
    data: {}
});

r.on('alertMe', function(){
	alert('Clicked!');
});
```

**Comment by evs-chris** [comment]
This is an unfortunate side-effect of how old-style events are parsed, which is why they are deprecated in 0.8 and removed in 0.9. The parsing for mustaches inside event directives is very touchy. I think this can be worked around by ignoring empty strings while processing tokens in the directive value, but it would be better to switch to a different construct.

Starting with 0.8, you can make the whole event directive conditional e.g. `<button {{#if true}}on-click="alertMe"{{/if}}>This also works</button>`

**Comment by frenchroasted** [comment]
Where is the write-up on the deprecation and removal of the "old-style events"? I looked through the 0.8 migration guide and the 0.9 changelog page and didn't see anything about removing "on-event='label'" syntax for attaching event handlers to nodes. While researching the problem I was having in my template, the only mention I could find about combining conditionals with event declarations was an old issue, I think, where it was mentioned that the preferred pattern was to do it inside the quotes. There is no mention of it in the API docs that I've seen, though I may be looking in the wrong place.

In any case, if the event handler declaration is changing for 0.9, that would be a huge undertaking for a large web application like the one I'm working on. What is the new syntax? How different is it from the old?

Meanwhile, the workaround suggested above is fine, but I would still point out that, for at least 0.8.x, a usage that breaks so completely as this one should generate some kind of parsing error. I would be equally happy with a code change that trimmed leading/trailing whitespace or else a parse error in this situation - makes no difference to me, I just think one of the two would be appropriate here. The current behavior is to silently fail, so anything would be an improvement.

**Comment by evs-chris** [comment]
Old-style events would be proxy events with arguments e.g. `on-click="foo:{{bar}},{{baz}}"`. The first entry under deprecations in the migration guide has the details for 0.8. The first breaking changes entry in 0.9 mentions the removal of argumented proxy events. The relevant issue at the root of these changes would be #1172, and there are a number of other issues and PRs linked from there.

The overall event mechanism is still `on-${event}="handler"`, it's just that what goes in `handler` has changed a bit. It is now either a single string to be proxied e.g. `on-click="someEvent"`, which will `fire` a `someEvent` event to be handled by `ractive.on('someEvent', ...)` or one or more expressions, which may be something like `on-click="@this.set('foo', 42)"` or `on-click="@this.fire('someEvent', event, 42)"` or `on-click="@this.toggle('show'), @this.fire('showing'), false"`. With 0.9, there's a reasonable approximation for argumented proxy events in `on-click="['someEvent', 42]"`, which is still just an expression, but the event directive special cases expressions that evaluate to an array with a string first element to handle conversion to the appropriate `fire` call.

**Comment by evs-chris** [comment]
This is now fixed in v0.8-dev and will go out with 0.8.12 when it is cut. Thanks for the report!
